### PR TITLE
Added godoc for exported blob client methods and types

### LINKED ISSUE #4: Functions, Types and Methods are undocumented
All the objects in the SDK packages are completely undocumented, making it hard to understand how to use the API. This is most obvious when reading the [godoc](http://godoc.org/github.com/MSOpenTech/azure-sdk-for-go/clients/storage) for the package.

### PR DESCRIPTION
**Pull request by ahmetalpbalkan**
Rearranged several methods like DeleteBlobIfExists and decided to not to export getProperties() methods.

Closes #4.

### CODE REVIEW

**Comment by ahmetalpbalkan** [comment]
cc: @paulmey

File: clients/storage/blob.go [hunk 1]
```diff
@@ -83,7 +95,7 @@ type ListContainersParameters struct {
 	Timeout    uint
 }
 
-func (p ListContainersParameters) GetParameters() url.Values {
+func (p ListContainersParameters) getParameters() url.Values {
```
Comment (paulmey) [inline clients/storage/blob.go]: someone may yell at you for pulling this from the exported symbols, but I agree with you

Comment (ahmetalpbalkan) [inline clients/storage/blob.go]: these should be used internally anyway

**Comment by paulmey** [comment]
LGTM

**Pull request merged**